space-framework/docs/space-framework.md

```diff
@@ -0,0 +1,92 @@
+# Space Framework
+
+## terminoloogy 
+**Physical Space**, **pSpace** A thing that behaves like a Kubernetes kube-apiserver (including the persistent storage behind it) and the subset of controllers in the kube-controller-manager that are concerned with API machinery generalities (not management of containerized workloads). A kcp logical cluster is an example. A regular Kubernetes cluster is another example.
+
+**Space Object**, **Space** - A kind of Kubernetes API object that _represents_ a pSpace.  Its "kind" is `Space`, we sometimes include the "object" part when writing specifically about the API objects to clearly distinguish them from what they represent.
+
+**Space Provider** - A Thing that manages the life cycle of multiple pSpaces. This includes at minimal create/delete/List operations on pSpaces. KIND is an example for Space Provider (that manages KIND clusters), KCP is another example - KCP manages KCP's workspaces. 
+
+## High level architecture  & Main componenets
+![SF overall view](SF-All.drawio.svg "SF overall viewgit s")
+
+The Space Framework (SF) is a generic management framework for space providers and pSpaces. The framework defines an abstraction layer for space providers and pSpace management that allows clients (both CLI/script based and client-go based clients) to use spaces while mantaining the clients decoupled from the specific pSpace and space provider that is being used. 
```

Comment:
Is there a CLI to the Space Framework?  I thought that there is only a kube-style API.  Of course, `kubectl` can be used to manipulate any kube API objects, that goes without saying.

Reply:
CLI here means kubectl   will explicitly say

Reply:
Perhaps replace that second sentence with something like the following.

> The framework defines an abstraction layer that is a common interface to space functionality. Clients of the abstraction layer interact with it rather than directly with the space providers. The abstraction layer's interface is a state-based interface expressed in Kubernetes API objects.